lib/deviantart_gallery_downloader.rb

```diff
@@ -220,7 +260,25 @@ def get_file_path(index, image_page_links, download_link, folder_name)
     file_name = download_link.split('?').first.split('/').last
     file_id = title_elem['href'].split('-').last
     file_ext = file_name.split('.').last
-    file_title = title.strip().gsub(/\.+$/, '').gsub(/^\.+/, '').strip().squeeze(" ").tr('/\\', '-')
+    file_title = title.strip().gsub(/\.+$/, '').gsub(/^\.+/, '').strip().squeeze(" ").tr('/:?\\', '-')
+
+    file_name = title_art+'-'+file_title+'.'+file_id+'.'+file_ext
+    file_path = "deviantart/#{@author_name}/#{@gallery_name}/#{folder_name}/#{file_name}"
+  end
+
+  def get_page_path(index, image_page_links, page_link, folder_name)
+    title_art_elem = @agent.page.parser.css(".dev-title-container h1 a")
+    title_elem = title_art_elem.first
+    title_art = title_art_elem.last.text
+    title = title_elem.text
+
+    puts "(#{index + 1}/#{image_page_links.count})Saving \"#{title}\""
+
+    #Sanitize filename
+    file_name = page_link.split('?').first.split('/').last
+    file_id = title_elem['href'].split('-').last
+    file_ext = 'htm'
+    file_title = title.strip().gsub(/\.+$/, '').gsub(/^\.+/, '').strip().squeeze(" ").tr('/:?\\', '-')
```

Comment:
Kinda repeat here, maybe the `file_title`, `file_name` and `file_path` should be methods.
```ruby
def sanitize_file_title(title)
  title.strip().gsub(/\.+$/, '').gsub(/^\.+/, '').strip().squeeze(" ").tr('/:?\\', '-')
end

file_title = sanitize_file_title(title)

# And so the name and path
```